src/custom/state/orders/hooks.ts

```diff
@@ -69,6 +69,12 @@ export const useOrders = ({ chainId }: GetOrdersParams): Order[] => {
   }, [state])
 }
 
+export const useAllOrders = ({ chainId }: GetOrdersParams) => {
+  const state = useSelector<AppState, OrdersState[ChainId] | undefined>(state => chainId && state.orders?.[chainId])
+
+  return state ? { ...state.pending, ...state.fulfilled } : {}
```

Comment:
No need to retrigger spread, return a new array each time, memoize

Reply:
inside useSelector is also fine yeah?

Reply:
Not exactly

`useSelector` works like [this](https://react-redux.js.org/7.1/api/hooks#useselector:~:text=When%20an%20action%20is%20dispatched%2C%20useSelector(),same%2C%20the%20component%20will%20not%20re%2Drender.). Basically when store changes, the internal subscriptions runs the selector callback and strict compares the result. If same,no rerender.
Therefore, if you create a new object inside selector callback -- always rerender

Still, memoization is helpful if something depends on the return of the whole hook, also micro-optimization for object spread, which is whatever

Reply:
nvm useMemo on return as your other examples do